tests/integration/targets/kinesis_stream/tasks/main.yml

```diff
@@ -0,0 +1,252 @@
+---
+- name: 'Setup AWS Module Defaults'
+  module_defaults:
+    group/aws:
+      aws_access_key: '{{ aws_access_key }}'
+      aws_secret_key: '{{ aws_secret_key }}'
+      security_token: '{{ security_token | default(omit) }}'
+      region: '{{ aws_region }}'
+    kinesis_stream:
+      # Number of shards is mandatory when state=present
+      shards: 1
+
+  block:
+  # ============================================================
+  # Set up some additional resources for later user
+
+#  # KMS Keys
+#  # Note: Because we're not a producer / consumer we don't actually need
+#  # access to the keys
+#  - name: 'Create KMS key 1'
+#    aws_kms:
+#      alias: '{{ kms_cmk_alias_1 }}'
+#      state: present
+#      enabled: yes
+#    register: create_kms_1
+#  - name: 'Create KMS key 2'
+#    aws_kms:
+#      alias: '{{ kms_cmk_alias_2 }}'
+#      state: present
+#      enabled: yes
+#    register: create_kms_2
+#  - name: 'Assert that we sucessfully created our keys'
+#    assert:
+#      that:
+#      - create_kms_1 is success
+#      - create_kms_2 is success
+#  - name: 'Store the Key IDs for later'
+#    set_fact:
+#        kms_cmk_id_1: '{{ create_kms_1.key_id }}'
+#        kms_cmk_arn_1: '{{ create_kms_1.key_arn }}'
+#        kms_cmk_id_2: '{{ create_kms_2.key_id }}'
+#        kms_cmk_arn_2: '{{ create_kms_2.key_arn }}'
+
+  # ============================================================
+  # Basic creation
+  - name: 'Create a basic Kinesis stream (CHECK_MODE)'
+    check_mode: yes
+    kinesis_stream:
+      name: '{{ kinesis_stream_name }}'
+    register: result
+  - name: 'Assert state is changed when first creating a stream (CHECK_MODE)'
+    assert:
+      that:
+      - result is success
+      - result is changed
+
+  - name: 'Create a basic Kinesis stream'
+    kinesis_stream:
+      name: '{{ kinesis_stream_name }}'
+    register: result
+  - name: 'Assert state is changed when first creating a stream'
+    assert:
+      that:
+      - result is success
+      - result is changed
+      - result.stream_arn.startswith('arn:aws:kinesis:')
+      - result.stream_arn.endswith(':stream/' + kinesis_stream_name)
+      - result.stream_status == 'ACTIVE'
+
+  - name: 'Create a basic Kinesis stream - Idempotency (CHECK_MODE)'
+    check_mode: yes
+    kinesis_stream:
+      name: '{{ kinesis_stream_name }}'
+    register: result
+  - name: 'Assert state is not changed when re-running the create (CHECK_MODE)'
+    assert:
+      that:
+      - result is success
+      - result is not changed
+    # XXX BUG
```

Comment:
Yeah, there's some bug in the existing check_mode logic. A perusal of the module doesn't immediately surface it.